app/controllers/hubstats/application_controller.rb

```diff
@@ -5,9 +5,14 @@ class ApplicationController < ApplicationController
 
     private
     def set_time
-      cookies[:hubstats_index] ||= 2
-      @start_date = DATE_RANGE_ARRAY[cookies[:hubstats_index].to_i][:date].ago.to_date
-      @end_date = Time.now
+      if cookies[:hubstats_index] == "null~~null" || cookies[:hubstats_index] == nil
```

Comment:
And we need the check if the index is null because that's what the index will be when there are no cookies. And both the application.js and the application controller need to check for both so that both the datepicker will show the right date and the data will all work correctly.